src/interactions/slashCommands/mixins/ApplicationCommandOptionBase.ts

```diff
@@ -0,0 +1,34 @@
+import type { APIApplicationCommandBasicOption, ApplicationCommandOptionType } from 'discord-api-types/v9';
+import { validateRequiredParameters, validateRequired } from '../Assertions';
+import { SharedNameAndDescription } from './NameAndDescription';
+
+export abstract class ApplicationCommandOptionBase<
```

Comment:
For this, rather than making `type` abstract wouldn't it be easier to have a constructor that takes the type and assigns it. That way the subclasses won't have to override types.

Reply:
No. This way the type is explicitly set to `Boolean` for instance instead of the generic `ApplicationCommandOptionType`